src/Domain/Services/CreateAccount.php

```diff
@@ -0,0 +1,233 @@
+<?php
+/**
+ * Service class implementing new create account behaviour for order processing.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\Domain\Services;
+
+use WP_REST_Request;
+use WC_Order;
+
+/**
+ * Service class implementing new create account behaviour for order processing.
+ */
+class CreateAccount {
+	/**
+	 * Instance of new account email class.
+	 *
+	 * @var WC_Email
+	 */
+	public $new_account_email = null;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+	}
+
+	/**
+	 * Init - register custom hook handler.
+	 */
+	public function init() {
+		// Register handler for signup during block-based checkout.
+		add_action( 'woocommerceblocks_create_account_for_order_request', [ $this, 'from_order_request' ], 10, 2 );
+
+		// Override core email handlers to add our new improved "new account" email.
+		add_action(
+			'woocommerce_email',
+			function ( $wc_emails_instance ) {
+				$this->new_account_email = require_once dirname( __DIR__ ) . '/class-wc-email-customer-new-account-blocks.php';
+
+				// Remove core "new account" handler; we are going to replace it.
+				remove_action( 'woocommerce_created_customer_notification', array( $wc_emails_instance, 'customer_new_account' ), 10, 3 );
+
+				// Add custom "new account" handler.
+				add_action(
+					'woocommerce_created_customer_notification',
+					function( $customer_id, $new_customer_data = array(), $password_generated = false ) use ( $wc_emails_instance ) {
+						// If this is a block-based signup, send a new email
+						// with password reset link (no password in email).
+						if ( isset( $new_customer_data['is_checkout_block_customer_signup'] ) ) {
```

Comment:
I don't see this key (`is_checkout_block_customer_signup`) being implemented anywhere?

Reply:
It's set in Checkout API - https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2851/files#diff-fcddc6e9639fa3e5ae3508597cb81054R210

Reply:
(if it wasn't, I would have discovered during testing!)

Reply:
doh, I missed that!